File: find-peak-element/find-peak-element.java
public class Solution {
    public int findPeakElement(int[] nums) {
        if (nums.length == 1) return 0;
        int l = 0, r = nums.length - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (mid == nums.length - 1 || nums[mid] > nums[mid + 1])
                r = mid - 1;
            else
                l = mid + 1;
        }
        return l;
    }
}

File: 162-find-peak-element/find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 0
        if len(nums) == 2:
            return nums.index(max(nums))
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            #ascending
            if (mid == 0 or nums[mid - 1] < nums[mid]) and mid + 1 < len(nums) and nums[mid + 1] > nums[mid]:
                left = mid + 1
            elif nums[mid - 1] > nums[mid]:
                right = mid - 1
            else:
                return mid